flink-connectors/flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/source/HBaseRowDataLookupFunction.java

```diff
@@ -68,54 +65,37 @@ public class HBaseRowDataLookupFunction extends TableFunction<RowData> {
     private transient HTable table;
     private transient HBaseSerde serde;
 
-    private final long cacheMaxSize;
-    private final long cacheExpireMs;
     private final int maxRetryTimes;
-    private transient Cache<Object, RowData> cache;
 
     public HBaseRowDataLookupFunction(
             Configuration configuration,
             String hTableName,
             HBaseTableSchema hbaseTableSchema,
             String nullStringLiteral,
-            HBaseLookupOptions lookupOptions) {
+            int maxRetryTimes) {
         this.serializedConfig = HBaseConfigurationUtil.serializeConfiguration(configuration);
         this.hTableName = hTableName;
         this.hbaseTableSchema = hbaseTableSchema;
         this.nullStringLiteral = nullStringLiteral;
-        this.cacheMaxSize = lookupOptions.getCacheMaxSize();
-        this.cacheExpireMs = lookupOptions.getCacheExpireMs();
-        this.maxRetryTimes = lookupOptions.getMaxRetryTimes();
+        this.maxRetryTimes = maxRetryTimes;
     }
 
     /**
      * The invoke entry point of lookup function.
      *
-     * @param rowKey the lookup key. Currently only support single rowkey.
+     * @param keyRow - A {@link RowData} that wraps lookup keys. Currently only support single
+     *     rowkey.
      */
-    public void eval(Object rowKey) throws IOException {
-        if (cache != null) {
-            RowData cacheRowData = cache.getIfPresent(rowKey);
-            if (cacheRowData != null) {
-                collect(cacheRowData);
-                return;
-            }
-        }
+    @Override
+    public Collection<RowData> lookup(RowData keyRow) throws IOException {
         for (int retry = 0; retry <= maxRetryTimes; retry++) {
             try {
                 // fetch result
-                Get get = serde.createGet(rowKey);
+                Get get = serde.createGet(((GenericRowData) keyRow).getField(0));
```

Comment:
`GenericRowData` is only one kind of implementation of `RowData`, the framework can ensure the keyRow is `GenericRowData` ?

Reply:
The implementation of LookupFunction will pass a GenericRowData as key row and it's safe to cast for now. I added a TODO here to update the logic once we improve the LookupFunction in the future.